src/battle_anim_effects_3.c

```diff
@@ -2325,6 +2325,88 @@ void AnimTask_SwallowDeformMon(u8 taskId)
     }
 }
 
+void AnimTask_MegaSwapSprite(u8 taskId)
```

Comment:
Since it's used for Mega/Ultra/Primal, maybe change the name to sth like `AnimTask_SmoothSpriteSwap`